src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -111,20 +124,23 @@ public Declaration createTree(Cursor c) {
     private Declaration createTreeInternal(Cursor c) {
         Position pos = CursorPosition.of(c);
         // dedup multiple declarations that point to the same source location
-        Declaration decl = declarationCache.get(pos);
-        if (decl == null) {
-            decl = switch (c.kind()) {
-                case EnumDecl -> createEnum(c);
-                case EnumConstantDecl -> createEnumConstant(c);
-                case FieldDecl -> createVar(c, Declaration.Variable.Kind.FIELD);
-                case ParmDecl -> createVar(c, Declaration.Variable.Kind.PARAMETER);
-                case FunctionDecl -> createFunction(c);
-                case StructDecl -> createRecord(c, Declaration.Scoped.Kind.STRUCT);
-                case UnionDecl -> createRecord(c, Declaration.Scoped.Kind.UNION);
-                case TypedefDecl -> createTypedef(c);
-                case VarDecl -> createVar(c, Declaration.Variable.Kind.GLOBAL);
-                default -> null; // skip
-            };
+        Optional<Declaration> cachedDecl = lookup(pos);
+        if (cachedDecl.isPresent()) {
+            return cachedDecl.get();
+        }
+        Declaration decl = switch (c.kind()) {
+            case EnumDecl -> createEnum(c);
+            case EnumConstantDecl -> createEnumConstant(c);
+            case FieldDecl -> createVar(c, Declaration.Variable.Kind.FIELD);
```

Comment:
FWIW, I see the `FieldDecl` case as being never executed in the code coverage report. Not sure why...

Reply:
I think this is caused by the fact that the new `TreeMaker::recordDeclaration` (introduced in previous PR) does all the processing of a struct/union inline.